move/axelar_gateway/sources/types/proof.move

```diff
@@ -60,6 +61,62 @@ public(package) fun recover_pub_key(
     ecdsa::secp256k1_ecrecover(&self.bytes, message, 0)
 }
 
+/// Validates the signatures of a message against the signers.
+/// The total weight of the signatures must be greater than or equal to the threshold.
+/// Otherwise, the error `ELowSignaturesWeight` is raised.
+public(package) fun validate(
+    self: &Proof,
+    message: vector<u8>,
+) {
+    let signers = self.signers();
+    let signatures = self.signatures();
+    assert!(signatures.length() != 0, ELowSignaturesWeight);
+
+    let threshold = signers.threshold();
+    let mut total_weight: u128 = 0;
+    let mut signer_index = 0;
+    let mut i = 0;
+
+    while(i < signatures.length()) {
+        let signature = signatures[i];
+
+        let pub_key = signature.recover_pub_key(&message);
+
+        let (weight, index) = find_weight_by_pub_key_from(signers, signer_index, &pub_key);
+
+        signer_index = index;
+
+        total_weight = total_weight + weight;
+
+        i = i + 1;
+
+        if (total_weight >= threshold) return
+    };
+
+    assert!(total_weight >= threshold, ELowSignaturesWeight);
```

Comment:
```suggestion
    abort ELowSignaturesWeight;
```
if this is reached, it's always insufficient. Double check the implementation matches the EVM gateway for sanity
https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/governance/BaseWeightedMultisig.sol

Reply:
i've addressed this implementation to match with the EVM gateway, but i'm not sure about [this check](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/febcd9fb2b4c0a6db23f5265f3ae9e115a3f2fc6/contracts/governance/BaseWeightedMultisig.sol#L219-L224) because it doesn't include in our [original implementation](https://github.com/axelarnetwork/axelar-cgp-sui/blob/479ddb615520f42eb25f7447522d1d1baa3cee94/move/axelar_gateway/sources/auth.move#L155-L192) as well.

Reply:
Yeah, that was an optional check added from a recent audit. Might consider removing it since it's overly restrictive, you can ignore it.